docs/source/theory.rst

```diff
@@ -4,5 +4,273 @@
 Theory
 ======
 
+--------------
+Bulk physics 
+--------------
 
-WIP
\ No newline at end of file
+H transport
+^^^^^^^^^^^
+
+The model developed by McNabb & Foster [:ref:`1 <McNabb and Foster>`] is used to model hydrogen transport in materials in FESTIM. The principle is to separate mobile hydrogen :math:`c_\mathrm{m}` and trapped hydrogen :math:`c_\mathrm{t}`. The diffusion of mobile particles is governed by Fick’s law of diffusion where the hydrogen flux is
+
+.. math::
+    :label: eq_difflux
+    
+    J = -D \nabla c_\mathrm{m}
+
+where :math:`D=D(T)` is the diffusivity. Each trap :math:`i` is associated with a 
+trapping and a detrapping rate :math:`k_i` and :math:`p_i`, respectively, as well 
+as a trap density :math:`n_i`.
+
+The temporal evolution of :math:`c_\mathrm{m}` and :math:`c_\mathrm{t}` are then 
+given by:
+
+.. math::
+    :label: eq_mobile_conc
+
+    \frac {\partial c_\mathrm{m}} { \partial t} = \nabla \cdot (D\nabla c_\mathrm{m}) - \sum \frac{\partial c_{\mathrm{t},i}}{\partial t} + \sum S_j
+
+
+.. math::
+    :label: eq_trapped_conc
+
+    \frac {\partial c_\mathrm{t}} { \partial t} = k_i c_\mathrm{m} (n_i - c_{\mathrm{t},i}) - p_i c_{\mathrm{t},i}
+
+where :math:`S_j=S_j(x,y,z,t)` is a source :math:`j` of mobile hydrogen. In FESTIM, 
+source terms can be space and time dependent. These are used to simulate plasma 
+implantation in materials, tritium generation from neutron interactions, etc. 
+These equations can be solved in cartesian coordinates but also in cylindrical 
+and spherical coordinates. This is useful for instance when simulating hydrogen 
+transport in a pipe or in a pebble. FESTIM can solve steady-state hydrogen transport 
+problems.
+
+
+Soret effect
+^^^^^^^^^^^^
+
+FESTIM can include the Soret effect [:ref:`2 <Longhurst>`] (also called
```

Comment:
We could also add [this](https://www.osti.gov/biblio/7333557) work to references.